src/call/mod.rs

```diff
@@ -683,4 +702,46 @@ impl SinkBase {
         self.batch_f.take();
         Poll::Ready(Ok(()))
     }
+
+    #[inline]
+    fn poll_flush<C: ShareCallHolder>(
+        &mut self,
+        cx: &mut Context,
+        call: &mut C,
+    ) -> Poll<Result<()>> {
+        if self.batch_f.is_some() {
+            ready!(self.poll_ready(cx)?);
+        }
+        if self.buf_flags.is_some() {
+            self.start_send_buffer_message(self.last_buf_hint, call)?;
```

Comment:
So the batch will not work again even after flush if it has been set to false in the history?

Reply:
Whether batch is enabled depends only on the current `enhance_buffer_strategy`

Reply:
No, once `last_buf_hint` becomes false, it will not become true again.

Reply:
you are right, but even if this hint is always false, it will not affect the batch, but it will affect the last message in the stream. (For example, if the user wants a stream of messages to be buffered, the last one will still not be bufferd)